packages/@react-aria/collections/src/BaseCollection.ts

```diff
@@ -208,4 +208,103 @@ export class BaseCollection<T> implements ICollection<Node<T>> {
     this.lastKey = lastKey;
     this.frozen = !isSSR;
   }
+
+  // TODO: this is pretty specific to menu, will need to check if it is generic enough
+  // Will need to handle varying levels I assume but will revisit after I get searchable menu working for base menu
+  // TODO: an alternative is to simply walk the collection and add all item nodes that match the filter and any sections/separators we encounter
+  // to an array, then walk that new array and fix all the next/Prev keys while adding them to the new collection
+  filter(filterFn: (nodeValue: string) => boolean): BaseCollection<T> {
```

Comment:
Definitely specific to menu, will need to update when we handle other collection components. Collections is in alpha so that should shield us from any changes that may need to happen to this function